cpp/src_prims/sparse/selection/knn.cuh

```diff
@@ -120,8 +120,9 @@ class sparse_knn_t {
                cudaStream_t stream_,
                size_t batch_size_index_ = 2 << 14,  // approx 1M
                size_t batch_size_query_ = 2 << 14,
-               ML::MetricType metric_ = ML::MetricType::METRIC_L2,
-               float metricArg_ = 0, bool expanded_form_ = false)
+               raft::distance::DistanceType metric_ =
+                 raft::distance::DistanceType::L2Unexpanded,
```

Comment:
Same, `L2SqrtUnexpanded` might be needed here.

Reply:
I would revert this to expanded form as well.